crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -175,6 +439,108 @@ impl<T: TransportConnect> Scheduler<T> {
         Ok(())
     }
 
+    fn choose_partition_processor_configuration(
+        nodes_config: &NodesConfiguration,
+        partition_replication: ReplicationProperty,
+        alive_workers: &NodeSet,
+        preferred_leader: Option<PlainNodeId>,
+        preferred_nodes: &NodeSet,
+    ) -> Option<PartitionConfiguration> {
+        let options = if let Some(preferred_leader) = preferred_leader {
+            NodeSetSelectorOptions::default().with_top_priority_node(preferred_leader)
+        } else {
+            NodeSetSelectorOptions::default()
+        }
+        .with_preferred_nodes(preferred_nodes);
+
+        // todo make nodeset selector select minimal sets ({zone: 2, node: 5} returning a nodeset of 5 nodes spread across 2 zones)
+        NodeSetSelector::select(
+            nodes_config,
+            &partition_replication,
+            |_node_id, node_config| node_config.has_role(Role::Worker),
+            |node_id, _node_config| alive_workers.contains(node_id),
```

Comment:
We should probably have something like a `WorkerState` stored in the `NodesConfiguration` which allows to mark a worker as draining so that partition processors are moved away from it automatically

Reply:
Done.